### PR TITLE
[yaml] Fix PubSub error message

### PR DESCRIPTION
**Pull request by Polber**
Adds a meaningful error message for `ReadFromPubSub` when a JSON/AVRO format is specified without a schema.

This PR also changes logic to allow upper or lowercase format similar to Java implementation.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by Polber** [comment]
R: @robertwb

File: sdks/python/apache_beam/yaml/yaml_io.py [hunk 1]
```diff
@@ -225,9 +229,11 @@ def _create_parser(
         schema_pb2.Schema(fields=[schemas.schema_field('payload', bytes)]),
         lambda payload: beam.Row(payload=payload))
   elif format == 'JSON':
+    _validate_schema()
     beam_schema = json_utils.json_schema_to_beam_schema(schema)
```
Comment (robertwb) [inline sdks/python/apache_beam/yaml/yaml_io.py:233]: Are the errors this raises on bad/unsupported schemas informative enough?

**Comment by robertwb** [comment]
Ping on this PR. It'd be good to close this usability gap.

**Review by robertwb — APPROVED** [review]
LGTM. This is a clear improvement, let's get this in before the cut.

**Pull request merged**